File: 03_jdbc/src/advance/Test07.java
/**
 *  Variable Argument
 *   : 값이 증가하거나 감소했을때 매개변수를 수정해줘야하는 것을 편리하게 만들자..
 */
package advance;

public class Test07 {
	/*
	private static void sum(int i, int j ) {
		System.out.println(i + j);
	}
	
	private static void sum(int i, int j, int k) {
		System.out.println(i + j + k);
	}
	private static void sum(int[] arr) {
		int result = 0;
		for (int val : arr) {
			result += val;
		}
		System.out.println(result);
	}
    */
	
	// ...  : Variable Argument 
	// int...arr : sum을 부를때 매개변수가 몇개오던 상관없이 쓸수있다.
	// 하지만 위치는 매개변수의 제일 마지막에 위치해야함.(앞이나 중간 X)
	private static void sum(int...arr) {
		int result = 0;
		for (int val : arr) {
			result += val;
		}
		System.out.println(result);
	}
	
	public static void main(String[] args) {
		sum(1, 10);
		sum(1, 10, 100);
		sum(1, 10, 100, 1000, 10000);
		sum(new int[] {1, 10});
		sum(new int[] {1, 10, 100});
	}
}

File: 10_javascript/02_function/test07.js
"use strict";

function sum(arr, start, end) {
    if (!arr) return 0;
    start = start || 0;
    end = end || arr.length;

    let result = 0;
    for (let i = start; i < end; i++) {
        if (typeof (arr[i]) != 'number') continue;
        result += arr[i];
    }
    return result;
}

let arr = [1, 2, '홍', 4, 5];
console.log( sum(arr, 0, arr.length) );  // 12
console.log( sum(arr, 0, 4) );           // 7
console.log( sum(arr, 0) );              // 12
console.log( sum(arr, 2) );              // 9
console.log( sum(arr) );                 // 12
console.log( sum() );                    // 0